notification_lib/novu_manager/notification_templates_manager.py

```diff
@@ -98,12 +98,16 @@ def update_novu_local_templates(cls) -> None:
         if not notification_group:
             NotificationGroupsManager.create_notification_group(name="General")
 
-        for template in generic_templates:
-            full_template = GenericNovuManager.get_generic_template_by_id(
-                template_id=template["id"]
-            )
-            # if it exists, it will override it to update, if it does not exist, it will create it
-            cls.create_update_notification_template(
-                template_name=full_template["template_name"],
-                steps=full_template["steps"],
-            )
+        # Check if every templates_to_update that were passed to the function exists among the generic templates.
+        # This way we only use the matching templates and not all of them.
+        for item in templates_to_update:
+            for generic_template_dict in generic_templates:
+                if item is not None and item in generic_template_dict['name']:
+                    full_template = GenericNovuManager.get_generic_template_by_id(
+                        template_id=generic_template_dict["id"]
+                    )
+                    # if it exists, it will override it to update, if it does not exist, it will create it
+                    cls.create_update_notification_template(
+                        template_name=full_template["template_name"],
+                        steps=full_template["steps"],
+                    )
```

Comment:
```suggestion
        for generic_template_dict in generic_templates:
                if generic_template_dict['name'] in templates_to_update:
                    full_template = GenericNovuManager.get_generic_template_by_id(
                        template_id=generic_template_dict["id"]
                    )
                    # if it exists, it will override it to update, if it does not exist, it will create it
                    cls.create_update_notification_template(
                        template_name=full_template["template_name"],
                        steps=full_template["steps"],
                    )
```